browser/js/userProfile/userProfile.js

```diff
@@ -0,0 +1,42 @@
+app.config(function($stateProvider) {
+
+    $stateProvider.state('userProfile', {
+        url: '/user-profile',
+        templateUrl: 'js/userProfile/userProfile.html',
+        controller: 'UserProfileCtrl',
+        resolve: {
+            user: function(AuthService) {
+                return AuthService.getLoggedInUser();
+            },
+            reviews: function(AuthService, ReviewFactory) {
+                return AuthService.getLoggedInUser().
+                then(function(user) {
+                    return ReviewFactory.getProductReviewsByUser(user._id);
+                });
+            },
+            orders: function(AuthService, OrderFactory) {
```

Comment:
Just pass in the user which gets resolved first, rather than doing the `AuthService` thing every time